src/components/dashboard/utils/sendFromQRCode.js

```diff
@@ -0,0 +1,39 @@
+// @flow
+import { NETWORK_ID } from '../../../lib/constants/network'
+
+export type CodeType = { networkId: string, address: string, amount: number }
+
+/**
+ * Returns a function that will decide where to navigate depending on the screen specified
+ * @param {screen} screen
+ * @returns {Function}
+ */
+export const sendFromQRCode = (screen: string): Function =>
```

Comment:
Seems like all use cases for this curried function the generated function can be decided on export or on the "curried" function execution. Maybe there is no need to generate functions inside the components for this?

Reply:
@dminones follow this suggestion and refactored the function. Now it receives two arguments and processes everything in the same function call.

Also updated tests.